tools/goctl/api/parser/g4/ast/apiparser.go

```diff
@@ -184,58 +207,40 @@ func (p *Parser) invoke(linePrefix, content string) (v *Api, err error) {
 	return
 }
 
-func (p *Parser) valid(mainApi, nestedApi *Api) error {
-	err := p.nestedApiCheck(mainApi, nestedApi)
-	if err != nil {
-		return err
-	}
-
-	mainHandlerMap := make(map[string]PlaceHolder)
-	mainRouteMap := make(map[string]PlaceHolder)
-	mainTypeMap := make(map[string]PlaceHolder)
-
-	routeMap := func(list []*ServiceRoute) (map[string]PlaceHolder, map[string]PlaceHolder) {
-		handlerMap := make(map[string]PlaceHolder)
-		routeMap := make(map[string]PlaceHolder)
-
+// storeVerificationInfo stores information for verification
+func (p *Parser) storeVerificationInfo(api *Api) {
```

Comment:
In this code block, the method `p.checkTypeDeclaration` should be called before put a `handler` & `route` into a map, by this way, the duplicate declarations error should be appeared and give a feedback to user.

Reply:
Hi, @anqiansong, p.valid has indeed been invoked in `apiparser.go` at line 175.